### DISCUSSION TITLE
Questions about seurat objects merging.

### DISCUSSION THREAD

**Discussion by Chap5732** [swaruplabUCI/ArchRtoSignac#30] [General]
I successfully obtain seurat objects of data from different time points of human embryos `PCW6, PCW7, PCW9, PCW10, PCW13, PCW16, PCW20, PCW22`  that are previously in ArchR objects format. But I am stuck at how I can merge these seurat objects in Signac to do integrative analysis given that each of them has different fixed width peak set. 

I am totally new in this field, so any word is welcomed here. Thanks.

**Comment by rootze** [comment]
@Chap5732 I am more than happy to help you out. Could you clarify some questions for me? Is your object in the Seurat format or ArchR? I am a bit confused which steps you are on if you are using the ArchRtoSignac package or you want to merge a list of objects?

**Comment by Chap5732** [comment]
Dear Ze,


Thanks for your concern, the data that I am working on is human prenatal
hypothalamus, to ascertain the chromatin accessibility landscape of
developmental hypothalamus, we dissected human embryos at different time
points(PCW6,PCW7,PCW9,PCW10,PCW13,PCW16,PCW20,PCW22) to get the
hypothalamus and harvested scATAC-seq data from it. There are 29 samples
from 8 time points. 10k cells are obtained after manual cleaning.

Not long ago I was a bit confused about whether I should use all fragment
file from 29 samples to create a big ArchR project and do the reduction and
cluster-based peak calling altogether and transform it to Seurat object or
should I just split them into 8 different time points first and create 8
ArchR project to do the transformation each and use merge function in
Signac to merge them. Now I know different ArchR project may have different
fixed-width peak set and the merge function in Signac may turn them in into
a union peak set with variable width. That's not what ArchRtoSignac is
meant for right? So I take the first way and I have done redcution,
cluster-based peak calling and Seurat tranformation with this big 10k-cell
ArchR projects.

Now I have got a new question, the Seurat object has peaks and RNA assay in
it but no tile matrix, so should I redo the reduction on this new peak
matrix and get a more accurate umap plot? (cause the previous one is done
with tile matrix in ArchR right?)

Again, very glad that you wrote and please feel free to tell me if you
don't understand anything that I have said.


Wish you the best,
Chap

Ze ***@***.***> 于2023年10月26日周四 12:40写道：

**Comment by Chap5732** [comment]
The total number of the cells is 100k, sorry for the mistake...

李国贤 ***@***.***> 于2023年10月26日周四 17:07写道：

**Comment by rootze** [comment]
Hello @Chap5732 , Yes, as you said you should use all fragment files from 29 samples to create a big ArchR project to do the reduction and cluster-based peak calling altogether and then transform it to a Seurat object, ideally using our `ArchRtoSignac` package. `ArchRtoSignac` is meant for data format conversion. It doesn't change anything that you did in the ArchR pipeline. And to answer your second question, the peak matrix in ArchR is almost the same as the tile matrix in ArchR. But extracting the peak matrix from ArchRObject is much easier and then converting it into a SeuratObject. I am unsure how you merge Seurat ...
> merge function in Signac


If you get peak matrix from ArchR, it should have a fixed-width peak set. And using ArchRtoSignac, you can keep it in your Seurat Object. I hope that make sense.

**Comment by Chap5732** [comment]
Hey Ze,


I met a new issue when I run getAnnotation() function. I succeeded last
time but this time I get an error in this code in both cluster and my own
laptop. Do you have any idea what's happening?

annotations <- getAnnotation(reference = EnsDb.Hsapiens.v86,
refversion = "hg38")[1] "In Progress:"
[1] "Extract genomic ranges from EnsDb object and prepare annotation"
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100%
elapsed=38s  Error in download.file(url, destfile, quiet = TRUE) :
  cannot open URL
'https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_assembly_report.txt'


Then I skip the getAnnotation() and try to use the previous preserved
one(Annotation.rds) to run the code and it runs for a while and cause error
in ArchR2Signac function.
[image: image.png]


Chap


Ze ***@***.***> 于2023年10月27日周五 06:06写道：

**Reply by rootze** [reply]
> Error in download.file(url, destfile, quiet = TRUE) : cannot open URL 'https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_assembly_report.txt'

Hello @Chap5732, thanks for using ArchRtoSignac. This is an issue; please report all issues to [issues section](https://github.com/swaruplabUCI/ArchRtoSignac/issues). But, since we are here, what I can say is this is not an error caused by the `ArchRtoSignac` package; it has something to do with problems in extracting reference info from NCBI. I don't know what you did before the two times you ran the code. Maybe your local R env changed.
